pkg/kubelet/cm/dra/plugin/dra_plugin_test.go

```diff
@@ -35,17 +35,28 @@ import (
 	drapbv1 "k8s.io/kubelet/pkg/apis/dra/v1"
 	drapbv1beta1 "k8s.io/kubelet/pkg/apis/dra/v1beta1"
 	"k8s.io/kubernetes/test/utils/ktesting"
+
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
 )
 
 type fakeGRPCServer struct {
 	drapbv1beta1.UnimplementedDRAPluginServer
 	drahealthv1alpha1.UnimplementedDRAResourceHealthServer
 	drapbv1.UnsafeDRAPluginServer
+
+	// if provided, the fake GRPC server will call prepare, or
+	// unprepare to return a custom response to the caller
+	prepare   func(ctx context.Context, in *drapbv1.NodePrepareResourcesRequest) (*drapbv1.NodePrepareResourcesResponse, error)
+	unprepare func(ctx context.Context, in *drapbv1.NodeUnprepareResourcesRequest) (*drapbv1.NodeUnprepareResourcesResponse, error)
 }
 
 var _ drapbv1.DRAPluginServer = &fakeGRPCServer{}
 
 func (f *fakeGRPCServer) NodePrepareResources(ctx context.Context, in *drapbv1.NodePrepareResourcesRequest) (*drapbv1.NodePrepareResourcesResponse, error) {
+	if f.prepare != nil {
+		return f.prepare(ctx, in)
+	}
```

Comment:
Would it be more clear to use custom fakeGRPCServer with redefined `NodePrepareResources` and `NodeUnprepareResources` and pass it to the `setupGRPCServerWithFake` than using this approach?